build.gradle

```diff
@@ -113,16 +120,41 @@ dependencies {
     }
 
     // These dependencies are only needed for running tests
-
-    testImplementation "org.junit.jupiter:junit-jupiter-api:5.6.2"
     testRuntimeOnly "org.junit.jupiter:junit-jupiter-engine:5.6.2"
+    testImplementation "org.junit.jupiter:junit-jupiter-api:5.6.2"
+
+    testImplementation 'org.hamcrest:hamcrest:2.2'
 
-    testImplementation group: 'org.mockito', name: 'mockito-inline', version: '3.3.3' // For final classes
+    testImplementation('org.spf4j:spf4j-slf4j-test:8.8.1') {
+        because "testable logging"
+    }
+
+    testImplementation("junit:junit:4.13")
+    testRuntimeOnly("org.junit.vintage:junit-vintage-engine:5.6.2") {
+        because " we haven't yet replaced spf4j-test Runner with a junit5 Extension"
+    }
+
+    testImplementation('org.mockito:mockito-inline:3.3.3') {
+        because "-inline build enables mocking final classes"
+        // https://javadoc.io/doc/org.mockito/mockito-core/latest/org/mockito/Mockito.html#0.2
+        // > Be aware that this artifact may be abolished when the inline mock making feature is integrated into the default mock maker.
+    }
+    testImplementation "org.mockito:mockito-junit-jupiter:3.3.3"
 
     // Config for our code analytics from: https://github.com/MovingBlocks/TeraConfig
     codeMetrics group: 'org.terasology.config', name: 'codemetrics', version: '1.4.0', ext: 'zip'
 }
 
+configurations.named("testRuntimeClasspath") {
+    gradlegoo.prefers(getResolutionStrategy(), "logging", "jcl-api-capability", "jcl-over-slf4j", "jcl should prefer slf4j when available")
```

Comment:
"java-commons logging", I think?

Reply:
Oh, I guess it would help to explain why this line is here too.

I don't have much opinion on java-commons-logging, it's not something we have a direct dependency on. But that component-metadata-plugin noticed there are several transitive dependencies that provide that capability, so it forces someone to make a choice.

Since most of the world seems to be operating on slf4j, when I saw that was one of the options it was trying to decide among, that seemed like the one to go with.